include/bout/coordinates.hxx

```diff
@@ -80,36 +87,46 @@ public:
    * Must be a better way so that Coordinates doesn't depend on Datafile
    */
   void outputVars(Datafile &file);
-  
-  Field2D dx, dy; ///< Mesh spacing in x and y
-  BoutReal dz; ///< Mesh spacing in Z
 
-  BoutReal zlength() const { return dz * nz; } ///< Length of the Z domain. Used for FFTs
+  FieldMetric dx, dy, dz; ///< Mesh spacing in x and y
+  // BoutReal dz; ///< Mesh spacing in Z
```

Comment:
I'd missed that `dz` is now always at least a `Field2D`. This is a breaking change we need to document in `CHANGELOG.md`.

I guess this is a tricky thing to provide a shim for, without basically writing our own full `double` replacement type. And the other option is lots of `#ifdef`s everywhere we use `dz`. It might be possible to have some thin wrapper with just `operator[]/()` to return the value.

It would be very nice if we can write an auto-upgrader for this change.